tests/tests_integration/test_api/test_datapoints.py

```diff
@@ -694,6 +695,127 @@ def test_numpy_dtypes_conversions_for_string_and_numeric(self, cognite_client, a
+        q2 = DatapointsQuery(external_id=bad_ts.external_id, treat_uncertain_as_bad=True, ignore_bad_datapoints=True)
+        q3 = DatapointsQuery(id=mixed_ts.id, treat_uncertain_as_bad=False, ignore_bad_datapoints=True)
+        q4 = DatapointsQuery(external_id=bad_ts.external_id, treat_uncertain_as_bad=False, ignore_bad_datapoints=True)
+        q5 = DatapointsQuery(id=mixed_ts.id, treat_uncertain_as_bad=False, ignore_bad_datapoints=False)
+        q6 = DatapointsQuery(external_id=bad_ts.external_id, treat_uncertain_as_bad=False, ignore_bad_datapoints=False)
+
+        for endpoint in retrieve_endpoints:
+            mix1, mix2, mix3, bad1, bad2, bad3 = endpoint(
+                id=[q1, q3, q5], external_id=[q2, q4, q6], include_status=False
+            )
+            assert mix1.status_code is mix1.status_symbol is bad1.status_code is bad1.status_symbol is None
+            assert len(mix1) == 117  # good
+            assert len(mix2) == 239  # good + uncertain
+            assert len(mix3) == 365  # good + uncertain + bad
+
+            assert len(bad1) == 0
+            assert len(bad2) == 0
+            assert len(bad3) == 365
+
+    def test_outside_points_with_bad_and_uncertain(self, retrieve_endpoints, ts_status_codes):
+        mixed_ts, bad_ts = ts_status_codes
+
+        for endpoint in retrieve_endpoints:
+            mix1, mix2, bad1, bad2, mix3, bad3, mix4, bad4 = endpoint(
+                id=[
+                    DatapointsQuery(id=mixed_ts.id, include_outside_points=False),
+                    DatapointsQuery(id=mixed_ts.id, include_outside_points=False, treat_uncertain_as_bad=False),
+                    DatapointsQuery(id=bad_ts.id, include_outside_points=False),
+                    DatapointsQuery(id=bad_ts.id, include_outside_points=False, ignore_bad_datapoints=False),
+                ],
+                external_id=[
+                    mixed_ts.external_id,
+                    bad_ts.external_id,
+                    DatapointsQuery(external_id=mixed_ts.external_id, treat_uncertain_as_bad=False),
+                    DatapointsQuery(external_id=bad_ts.external_id, ignore_bad_datapoints=False),
+                ],
+                include_outside_points=True,
+                start=ts_to_ms("2023-01-10") + 1,  # first good dp
+                end=ts_to_ms("2023-12-27"),  # last good dp
+                include_status=False,
+                treat_uncertain_as_bad=True,
+                ignore_bad_datapoints=True,
+            )
+            assert len(mix1) == 115  # good only, no outside
+            assert len(mix3) == 117  # good only, with outside
+            assert len(mix2) == 233  # good+uncertain, no outside
+            assert len(mix4) == 235  # good+uncertain, with outside
+
+            assert len(bad1) == 0
+            assert len(bad3) == 0
+            assert len(bad2) == 350
+            assert len(bad4) == 352
+
+    def test_status_codes_and_symbols(self, retrieve_endpoints, ts_status_codes):
```

Comment:
Shouldn't there also be a test of aggregating and including/not including uncertain/bad datapoints?

Or is aggregates not included in this PR?

Reply:
Agreed, will add!